lib/pure/os.nim

```diff
@@ -702,6 +702,8 @@ proc getAppFilename*(): string {.rtl, extern: "nos$1", tags: [ReadIOEffect], noW
       result = getApplHaiku()
     elif defined(openbsd):
       result = getApplOpenBsd()
+    elif defined(nintendoswitch):
+      result = "nxNimApp"
```

Comment:
why are you doing this? The path of the nro is passed in via argv[0] by the hbmenu.

Reply:
also, devkitpro didn't mention `cmdCount` in any of its files :(

Reply:
it used to work (I fixed it myself in a commit some time ago), though looks like it's broken now.

cmdCount and cmdLine are from Nim: https://github.com/nim-lang/Nim/blob/1431f90d8c3d3ed0afcaee62073d5808cea227ff/compiler/cgen.nim#L1411

Reply:
So how about stubbing it for now in order to make compilation work and migrate onto working solution under different PR. 

I don't think amount of work will increase much if we do it later :)

Reply:
Thanks for pointing out that cmdLine is nim-related, BTW

Reply:
If you agree to hold this fix for later, would you pls resolve this thread? :)

Reply:
yes, though I'd remove the change you did in this PR.

Reply:
If i remove this change it will freeze in a hard-to-debug locations that I used to find using echo for a couple of hours, so I'd really like to keep the stuff :D

Reply:
yeah but don't hardcode the name of your application into Nim.

You can use GDB on Switch btw, https://gist.github.com/jam1garner/c9ba6c0cff150f1a2480d0c18ff05e33

Reply:
Fair enough, I'll try to fix cmdLine / cmdCount or remove the temp fix completely. 

I tried to pick a generic name: nx stands switch, Nim is surely being used, and App is, well, app.

Reply:
A-ha, the issue goes far than I thought.

It's because I use `--noMaim` option, so these params are turned off. 

Should I fix it in logging, so it would not depend on this functionality?

Reply:
Okay, so I used the getApplHeuristic that feels like it suits my case exactly

Reply:
@RSDuck, can you please review my changes when you have time and ether ask for more changes or resolve this conversation? :)